### PR TITLE
2024.01 changelog and version updates

### PR DESCRIPTION
**Pull request by rem1776**
**Description**
2024.01 updates

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,61 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
...
+
+### Known Issues
+- Diag Manager Rewrite:
+	- If two empty files are present in the diag_table.yaml file the code will crash with a allocation error (#1506)
+	- Setting an output frequency of '0 days' does not work as expected and may cause an error stating a time_step has been skipped (#1502)
+	- The `flush_nc_files` and `mix_snapshot_average_fields` nml options are not yet functional. The `mix_snapshot_average_fields` option is planned will to be deprecated (for the rewritten diag_manager only).
+	- Expected output file changes:
+		- If the model run time is less than the output frequency, old diag_manager would write a specific value (9.96921e+36). The new diag_manager will not, so only fill values will be present.
+		- A `scalar_axis` dimension will not be added to scalar variables
+		- The `_FillValue` and `missing_value` attributes will not be added for `average_T1`, `average_T2`, and `average_DT` variables
```
Comment (uramirez8707) [inline CHANGELOG.md:19]: This should actually say that the average_* variable will no longer be added as they are non-standard conventions.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,61 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
 `rr` is a sequential release number (starting from `01`), and an optional two-digit
 sequential patch number (starting from `01`).
 
+## [2024.01] - 2024-05-03
+
+### Known Issues
+- Diag Manager Rewrite:
+	- If two empty files are present in the diag_table.yaml file the code will crash with a allocation error (#1506)
+	- Setting an output frequency of '0 days' does not work as expected and may cause an error stating a time_step has been skipped (#1502)
+	- The `flush_nc_files` and `mix_snapshot_average_fields` nml options are not yet functional. The `mix_snapshot_average_fields` option is planned will to be deprecated (for the rewritten diag_manager only).
```
Comment (bensonr) [inline CHANGELOG.md:15]: Simple typo here of "is planned will to be"

**Comment by rem1776** [comment]
@bensonr @uramirez8707 This should be ready now, it's been updated for those two comments

**Review by uramirez8707 — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,61 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
...
+- BUILD(HDF5): HDF5 version 1.14.3 generates floating point exceptions, and will cause errors if FMS is built with FPE traps enabled.
+
+### Added
+- DIAG_MANAGER: The diag manager has been rewritten with a object oriented design. The old diag_manager code has been kept intact and will be used by default. The rewritten diag manager can be enabled via `use_modern_diag = .true.` to your `diag_manager_nml`. New features include:
+	- Self-describing YAML formatting for diag_table's
+	- Allows 4d variables
+  - Support defining subregions with indices
+  - More flexibility when adding metadata and defining output frequency
+- FMS2_IO: Adds support for collective parallel reads to improve model startup time. The collective reads are disabled by default and enabled via the `use_collective` flag in `netcdf_io_mod`.
+- DATA_OVERRIDE: Adds multifile support for using 3 input netcdf files instead of one. Three keys have been added to the data_table: `is_multi_file` to be set to true to enable the feature, as well as `prev_file_name` and `next_file_name` to set to the names of the additional files.
```
Comment (bensonr) [inline CHANGELOG.md:35]: The multi-file support in data_override is only up to 3 files?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,61 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
...
+- INTERPOLATOR: Adds support for yearly/annual data
+- DATA_OVERRIDE: Adds support for monotonically increasing/decreasing arrays
```
Comment (bensonr) [inline CHANGELOG.md:37]: I think this needs clarified for which axes we are able to do this for or at a minimum a reference to the GitHub issue and/or PR.

**Review by bensonr — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,61 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
...
+- DIAG_MANAGER: When using the `do_diag_field_log` nml option, the output log file may be ovewritten if using a multiple root pe's
+- TESTS: `test_mpp_gatscat.F90` fails to compile with the Intel Oneapi 2024.01's version of ifort
+- BUILD(HDF5): HDF5 version 1.14.3 generates floating point exceptions, and will cause errors if FMS is built with FPE traps enabled.
+
+### Added
+- DIAG_MANAGER: The diag manager has been rewritten with a object oriented design. The old diag_manager code has been kept intact and will be used by default. The rewritten diag manager can be enabled via `use_modern_diag = .true.` to your `diag_manager_nml`. New features include:
+	- Self-describing YAML formatting for diag_table's
+	- Allows 4d variables
+  - Support defining subregions with indices
+  - More flexibility when adding metadata and defining output frequency
```
Comment (thomas-robinson) [inline CHANGELOG.md:33]: - Updated math and buffer handling for better support when send_data is called in a threaded region 
- New diag manager requires a call to diag send complete.  
- @uramirez8707 Do you think there's anything else to add?

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,61 @@ and this project uses `yyyy.rr[.pp]`, where `yyyy` is the year a patch is releas
...
+		- Subregional output will have a global attribute `is_subregional = True` set for non-global history files.
+		- The `grid_type` and `grid_tile` global attributes will no longer be added for all files, and some differences may be seen in the exact order of the `associated_files` attribute
+
+- DIAG_MANAGER: When using the `do_diag_field_log` nml option, the output log file may be ovewritten if using a multiple root pe's
+- TESTS: `test_mpp_gatscat.F90` fails to compile with the Intel Oneapi 2024.01's version of ifort
+- BUILD(HDF5): HDF5 version 1.14.3 generates floating point exceptions, and will cause errors if FMS is built with FPE traps enabled.
+
+### Added
+- DIAG_MANAGER: The diag manager has been rewritten with a object oriented design. The old diag_manager code has been kept intact and will be used by default. The rewritten diag manager can be enabled via `use_modern_diag = .true.` to your `diag_manager_nml`. New features include:
+	- Self-describing YAML formatting for diag_table's
```
Comment (thomas-robinson) [inline CHANGELOG.md:30]: You dont need the apostrophe in diag_tables

**Review by thomas-robinson — CHANGES_REQUESTED** [review]

**Pull request merged**